### PR TITLE
Add EDN-based external Schema DSL (#9)

### PR DESCRIPTION
**Pull request by cblp**
This change is [Reviewable](https://reviewable.io/reviews/ff-notes/ron/32)

### CODE REVIEW

File: ron/lib/RON/Schema/EDN.hs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        Nothing ->
+            put env { knownTypes =
+                        Map.insert structName (structLww struct) knownTypes
+                    }
+        Just _ ->
+            fail $ "duplicate declaration of type " ++ Text.unpack structName
+    pure struct
+
+  where
+
```
Comment (willbasky) [inline ron/lib/RON/Schema/EDN.hs]: Did empty line rest for a comment?

Comment (cblp) [reply ron/lib/RON/Schema/EDN.hs]: No, they are for the visual balance with empty lines between functions inside `where` block.

File: ron/lib/RON/Schema/EDN.hs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    parseName =
+        withNoTag $
+        withSymbol "struct_lww name symbol" $
+        withNoPrefix "struct_lww name" $
+        pure . Text.decodeUtf8
+
+    parseFields = \case
+        [] -> pure mempty
+        nameTV : typeTV : cont -> do
```
Comment (willbasky) [inline ron/lib/RON/Schema/EDN.hs]: TV == Tagged Value?

File: ron/lib/RON/Schema/EDN.hs (hunk 1 above)
Comment (willbasky) [reply ron/lib/RON/Schema/EDN.hs]: I offer to leave an empty line  before `where`, not after one. Here and feather.

File: ron/lib/RON/Schema/EDN.hs (hunk 2 above)
Comment (willbasky) [reply ron/lib/RON/Schema/EDN.hs]: May be `taggedName` and so on?

File: ron/lib/RON/Schema/EDN.hs (hunk 1 above)
Comment (cblp) [reply ron/lib/RON/Schema/EDN.hs]: I don't like it. Maybe vice versa? Anyway, bikeshedding.

**Pull request merged**